pennylane_lightning/core/lightning_base.py

```diff
@@ -348,22 +348,22 @@ def _check_adjdiff_supported_measurements(measurements: List[MeasurementProcess]
         if not measurements:
             return None
 
-        if len(measurements) == 1 and measurements[0].return_type is State:
+        if len(measurements) == 1 and isinstance(measurements[0], StateMP):
             # return State
             raise qml.QuantumFunctionError(
                 "Adjoint differentiation does not support State measurements."
             )
 
         # The return_type of measurement processes must be expectation
-        if any(m.return_type is not Expectation for m in measurements):
+        if any(not isinstance(m, ExpectationMP) for m in measurements):
             raise qml.QuantumFunctionError(
                 "Adjoint differentiation method does not support expectation return type "
```

Comment:
something like this?
```suggestion
                "Adjoint differentiation method does not support Expectation Measurement type "
```